modules/batch-tool/src/main/java/com/devonfw/module/batch/common/cli/AbstractJobSubCommand.java

```diff
@@ -0,0 +1,107 @@
+package com.devonfw.module.batch.common.cli;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.springframework.batch.core.BatchStatus;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobInstance;
+import org.springframework.batch.core.explore.JobExplorer;
+import org.springframework.batch.core.repository.JobRepository;
+
+abstract class AbstractJobSubCommand {
+
+  @Inject
+  protected JobRepository jobRepository;
+
+  @Inject
+  private JobExplorer jobExplorer;
+
+  private Long getLongIdentifier(String jobIdentifier) {
+
+    try {
+      return new Long(jobIdentifier);
+    } catch (NumberFormatException e) {
+      // Not an ID - must be a name
+      return null;
+    }
+  }
+
+  /**
+   * @param jobIdentifier a job execution id or job name
+   * @param minStatus the highest status to exclude from the result
+   * @return
```

Comment:
great that you provide helpful JavaDoc. Can you even complete it and fill `@return` (I now it sucks, but if we have the tags present, we should also add something useful to it - just write "the {@link List} or matching {@link JobExecution}s." and JavaDoc is complete and all happy).

Reply:
see above, that's not my javadoc ;-)

Reply:
See above. I have to reject this PR if you infridge CLA or copyright issues.
Please not that devon4net repo has been entirely wiped due to some tiny public OSS issue. They lost all their issues, PRs and git history for a stupid small mistake. Lets take such things very serious. People are watching us and we are happy that we can do this as OSS and on github. Small mistakes can ruin all this...

Reply:
I simplified and thereby totally rewrote the code here